### PR TITLE
feat(question-type-cursor): #FOR-504 add rules for cursor's choices

### PR DESCRIPTION
**Pull request by c4role**
## Describe your changes
Add new rules to check if min and max values are consistent with step choice.

## Checklist tests
Create a question type cursor, choose min and max val of your choice and add a step. If it's not consistent (exemple if you choose min val = 0 & max val = 10, your step must be peer)

## Issue ticket number and link
[FOR-504](https://jira.support-ent.fr/browse/FOR-504)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by simonchaval — APPROVED** [review]

File: common/src/main/resources/ts/models/FormElement/FormElement.ts [hunk 1]
```diff
@@ -7,6 +7,10 @@ export abstract class FormElement implements Selectable {
     position: number;
     nb_responses: number;
     selected: boolean;
+    question_type: number;
+    cursor_min_val: number;
+    cursor_max_val: number;
+    cursor_step: number;
```
Comment (Sofianebernoussi) [inline common/src/main/resources/ts/models/FormElement/FormElement.ts:13]: peut être ca peut être cool de modeliser les cursors ? En faire un objet avec les trois attributs : min, max, step ?

File: formulaire/src/main/resources/i18n/en.json [hunk 2]
```diff
@@ -240,6 +240,7 @@
   "formulaire.question.cursor.default.min.val": "1",
   "formulaire.question.cursor.default.max.val": "10",
   "formulaire.question.cursor.default.step": "1",
+  "formulaire.question.cursor.inconsistency.between.values": "Careful, the step must be a multiple of the difference between the maximum value and the minimum value of the cursor.",
```
Comment (Sofianebernoussi) [inline formulaire/src/main/resources/i18n/en.json:243]: je sais aps si pour formulaire vous devez pas faire toutes les autres langues ? pt, es, de etc..

File: formulaire/src/main/resources/public/ts/directives/question/question-item.ts [hunk 3]
```diff
@@ -144,6 +150,14 @@ export const questionItem: Directive = ng.directive('questionItem', () => {
                 vm.question.mandatory = vm.question.conditional;
                 $scope.$apply();
             };
+
+            vm.cursorChoiceIsConsistent = () : boolean => {
+                let consistentChoice = (vm.question.cursor_max_val - vm.question.cursor_min_val) % vm.question.cursor_step == 0;
```
Comment (Sofianebernoussi) [inline formulaire/src/main/resources/public/ts/directives/question/question-item.ts:155]: du coup si tu le modelistes tu pourras faire ce genre d'opération dans le model par exemple !

**Review by Sofianebernoussi — APPROVED** [review]

File: formulaire/src/main/resources/i18n/en.json (hunk 2 above)
Comment (c4role) [reply formulaire/src/main/resources/i18n/en.json:243]: Nope pas à ma connaissance

**Pull request merged**